### PR TITLE
merge by rowData

### PR DESCRIPTION
**Pull request by 0xMuluh**
see #397 and #400

### CODE REVIEW

File: R/merge.R [hunk 1]
```diff
@@ -103,6 +103,16 @@ setGeneric("mergeSamples",
            function(x, f, archetype = 1L, ...)
                standardGeneric("mergeSamples"))
 
+# Helper function to check if 'f' is a valid column name in rowData or colData
+.is_valid_column_name <- function(x, f, dim_type) {
+    if (dim_type == "rows") {
```
Comment (antagomir) [inline R/merge.R]: by the way could we easilyl use `dim.type` instead of `dim_type`?

Comment (antagomir) [reply R/merge.R]: Why we don't here use MARGIN (as usual for rows vs. columns; or features and samples)? See other functions for such cases. Consistency could be desirable.

**Review by antagomir — COMMENTED** [review]
Seems good but see the comment on argument name. 

Checks still fail.

File: R/merge.R (hunk 1 above)
Comment (0xMuluh) [reply R/merge.R]: Oh this function is removed. I was testing several approaches.

**Comment by 0xMuluh** [comment]
> Seems good but see the comment on argument name.
> 
> Checks still fail.

I'm not sure about the fail, but test run fine on local. will check it out

**Comment by antagomir** [comment]
up

**Comment by antagomir** [comment]
ready?

**Comment by 0xMuluh** [comment]
> ready?

Yes.

**Comment by antagomir** [comment]
Great. Are we now ready to address
https://github.com/FrederickHuangLin/ANCOMBC/issues/174

**Comment by 0xMuluh** [comment]
> Great. Are we now ready to address [FrederickHuangLin/ANCOMBC#174](https://github.com/FrederickHuangLin/ANCOMBC/issues/174)

Ohh yea.  Would make a PR

**Comment by antagomir** [comment]
Status of this one?

**Pull request merged**